test/framework/src/main/java/org/opensearch/test/rest/yaml/section/MatchAssertion.java

```diff
@@ -110,6 +111,28 @@ protected void doAssert(Object actualValue, Object expectedValue) {
         }
 
         if (expectedValue.equals(actualValue) == false) {
+            if (expectedValue instanceof ArrayList && actualValue instanceof ArrayList) {
```

Comment:
You don't need to workaround the comparison, please use index based comparison (instead of array):

```
- match: { hits.hits.0.sort: [7] }
```

Would become

```
- match: { hits.hits.0.sort.0: 7 }
```

And works out of the box